Gems/Atom/Tools/AtomToolsFramework/Code/Tests/ViewportInteractionImplTests.cpp

```diff
@@ -0,0 +1,171 @@
+        {
+            const auto halfScreenDimensions = ScreenDimensions * 0.5f;
+            return AzFramework::ScreenPoint(halfScreenDimensions.m_width, halfScreenDimensions.m_height);
+        }
+
+        void SetUp() override
+        {
+            AZ::NameDictionary::Create();
+
+            m_view = AZ::RPI::View::CreateView(AZ::Name("TestView"), AZ::RPI::View::UsageCamera);
+
+            const auto aspectRatio = aznumeric_cast<float>(ScreenDimensions.m_width) / aznumeric_cast<float>(ScreenDimensions.m_height);
+
+            AZ::Matrix4x4 viewToClipMatrix;
+            AZ::MakePerspectiveFovMatrixRH(viewToClipMatrix, AZ::DegToRad(60.0f), aspectRatio, 0.1f, 1000.f, true);
+            m_view->SetViewToClipMatrix(viewToClipMatrix);
+
+            m_viewportInteractionImpl = AZStd::make_unique<AtomToolsFramework::ViewportInteractionImpl>(m_view);
+
+            m_viewportInteractionImpl->m_deviceScalingFactorFn = []
+            {
+                return 1.0f;
+            };
+            m_viewportInteractionImpl->m_screenSizeFn = []
+            {
+                return ScreenDimensions;
+            };
+
+            m_viewportInteractionImpl->Connect(TestViewportId);
+        }
+
+        void TearDown() override
+        {
+            m_viewportInteractionImpl->Disconnect();
+            m_viewportInteractionImpl.reset();
+
+            m_view.reset();
+
+            AZ::NameDictionary::Destroy();
+        }
+
+        AZ::RPI::ViewPtr m_view;
+        AZStd::unique_ptr<AtomToolsFramework::ViewportInteractionImpl> m_viewportInteractionImpl;
+    };
+
+    // transform a point from screen space to world space, and then from world space back to screen space
+    AzFramework::ScreenPoint ScreenToWorldToScreen(
+        const AzFramework::ScreenPoint& screenPoint,
+        AzToolsFramework::ViewportInteraction::ViewportInteractionRequests& viewportInteractionRequests)
+    {
+        const auto worldResult = viewportInteractionRequests.ViewportScreenToWorld(screenPoint);
+        return viewportInteractionRequests.ViewportWorldToScreen(worldResult);
+    }
+
+    TEST_F(ViewportInteractionImplFixture, ViewportInteractionRequestsMapsFromScreenToWorldAndBack)
+    {
+        using AzFramework::ScreenPoint;
+
+        m_view->SetCameraTransform(AZ::Matrix3x4::CreateFromMatrix3x3AndTranslation(
+            AZ::Matrix3x3::CreateRotationZ(AZ::DegToRad(90.0f)), AZ::Vector3(10.0f, 0.0f, 5.0f)));
+
+        {
+            const auto expectedScreenPoint = ScreenPoint{ 600, 450 };
+            const auto resultScreenPoint = ScreenToWorldToScreen(expectedScreenPoint, *m_viewportInteractionImpl);
+            EXPECT_EQ(resultScreenPoint, expectedScreenPoint);
+        }
+
+        {
+            auto expectedScreenPoint = ScreenCenter();
+            const auto resultScreenPoint = ScreenToWorldToScreen(expectedScreenPoint, *m_viewportInteractionImpl);
+            EXPECT_EQ(resultScreenPoint, expectedScreenPoint);
+        }
```

Comment:
Nit, repeated: I always prefer to have tests call the function under test only once. If you have multiple data sets that you want to test, use a parameterized fixture.

Reply:
Hmm I do most of the time but here it's not a deal breaker for me, I could rewrite this as a parameterized test but I think that add more noise to what's going on. I'd be inclined to leave this as-is for now, and if we add more tests like this in future it's easy to update and refactor